File: app/src/main/java/com/youer/genshin/ui/BaseFragment.java
package com.youer.genshin.ui;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import com.youer.genshin.constants.Constants;

/**
 * @author youer
 * @date 9/20/23
 */
public class BaseFragment extends Fragment {

    Long uid;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle args = getArguments();
        if (args != null) {
            uid = args.getLong(Constants.KEY_UID);
        }
    }

}

File: app/src/main/java/com/youer/genshin/ui/BaseFragment.kt
package com.youer.genshin.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.youer.genshin.constants.Constants
import com.youer.genshin.presenter.MainPresenter

/**
 * @author youer
 * @date 9/20/23
 */
abstract class BaseFragment(presenter:MainPresenter) : Fragment() {
    var uid: Long? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.apply {
            uid = getLong(Constants.KEY_UID)
        }
    }
}